lib/ddtrace/contrib/extensions.rb

```diff
@@ -140,4 +148,7 @@ def reduce_log_verbosity
       end
     end
   end
+
+  # Load and extend Contrib by default
+  extend Contrib::Extensions
 end
```

Comment:
This is a tricky one: ideally `lib/ddtrace.rb` doesn't need to know that contrib exists.
Contrib is additive to the core library, and configures itself as such (or at least that's what we wanted).

Some parts are more core than other, and can't be feasibly separated, but for contrib that was the goal with having it all under `lib/ddtrace/contrib`.

There's one point where the "Datadog product" is created, bringing this all together and breaking this separation, but I was trying to avoid it if possible here.

Reply:
I guess this is part of a bigger conversation, but I'm not a fan at all of the of modules deciding to add themselves to our classes at random points, as it makes it really hard to reason about and follow the codebase.

I think the registry approach we use is actually a much cleaner option: yes, one still needs to find every caller of "add_to_registry", but it becomes absolutely clear that we have such a pattern, and the class managing the registry keeps control (e.g. it can decide what happens when "add_to_registry" is called), rather than with modules where at most you can get a callback from Ruby after the fact.

Failing that, if this behavior is loaded by default when `require 'ddtrace'` gets called, then I still believe it should be on `datadog.rb`. That way, we have one file we can examine to determine what does in there, and we don't need to actually start Ruby in order to instrospect what ends up in there 😄

Reply:
I agree that a registry type of thing would be best.

I don't think we have clear vision on all the APIs required to create a plugin system that works with contrib/profiling/security/ci today, but I agree that a well-defined registry for it would be best.